pkg/utils/registry/registry.go

```diff
@@ -39,3 +47,40 @@ func ParseImage(imageID string) (Image, error) {
 		Repo: repo,
 	}, nil
 }
+
+func ParseLabels(labels map[string]string) (rRef, rDigest string) {
+	if ref, ok := labels[label.ImageRef]; ok {
+		rRef = ref
+	}
+	if layerDigest, ok := labels[label.CRIDigest]; ok {
+		rDigest = layerDigest
+	}
+	return
+}
+
+func AuthnTransport(ref name.Reference, tr http.RoundTripper, keychain authn.Keychain) (http.RoundTripper, error) {
+	if keychain == nil || (reflect.ValueOf(keychain).Kind() == reflect.Ptr && reflect.ValueOf(keychain).IsNil()) {
```

Comment:
Why we need to check `nil` interface?

Reply:
When no auth exists, a null pointer will be passed in from the outside, which will cause a non-nil interface to exist.